src/smtencoding/FStar.SMTEncoding.Term.fs

```diff
@@ -1013,18 +1013,10 @@ and mkPrelude z3options =
                                  (fst boxIntFun,     [snd boxIntFun,  Int_sort, true],   Term_sort, 7, true);
                                  (fst boxBoolFun,    [snd boxBoolFun, Bool_sort, true],  Term_sort, 8, true);
                                  (fst boxStringFun,  [snd boxStringFun, String_sort, true], Term_sort, 9, true);
-                                 (fst boxRealFun,    [snd boxRealFun, Sort "Real", true], Term_sort, 10, true);
-                                 ("LexCons",    [("LexCons_0", Term_sort, true); ("LexCons_1", Term_sort, true); ("LexCons_2", Term_sort, true)], Term_sort, 11, true)] in
+                                 (fst boxRealFun,    [snd boxRealFun, Sort "Real", true], Term_sort, 10, true)] in
    let bcons = constrs |> List.collect (constructor_to_decl norng)
                        |> List.map (declToSmt z3options) |> String.concat "\n" in
-   let lex_ordering = "\n(define-fun is-Prims.LexCons ((t Term)) Bool \n\
-                                   (is-LexCons t))\n\
-                       (declare-fun Prims.lex_t () Term)\n\
-                       (assert (forall ((t1 Term) (t2 Term) (x1 Term) (x2 Term) (y1 Term) (y2 Term))\n\
-                                    (iff (Valid (Prims.precedes Prims.lex_t Prims.lex_t (LexCons t1 x1 x2) (LexCons t2 y1 y2)))\n\
-                                         (or (Valid (Prims.precedes t1 t2 x1 y1))\n\
-                                             (and (= x1 y1)\n\
-                                                  (Valid (Prims.precedes Prims.lex_t Prims.lex_t x2 y2)))))))\n\
+   let lex_ordering = "\n(declare-fun Prims.lex_t () Term)\n\
```

Comment:
Also, note that the smtencoding still has a `lex_t` type, though it is not exposed to the user. Basically, we emit queries for `precedes t1 t2 x y`, and switch to `precedes lex_t lex_t x y` as before.

Reply:
@nikswamy

Reply:
I think this lex_t is just a dummy type constant at this point. Maybe we can just remove it? And use Prims.unit for this purpose instead?